shared-libs/phone-number/src/phone-number.js

```diff
@@ -34,6 +35,9 @@ var _init = function(settings, phone) {
           scheme = phonenumber.PhoneNumberFormat.INTERNATIONAL;
         }
       }
+      if (shortInfo.isValidShortNumber(parsed)) {
+        scheme = phonenumber.PhoneNumberFormat.NATIONAL;
+      }
```

Comment:
If this is going to overwrite the scheme regardless then we shouldn't bother doing lines 31-37. Consider...

```js
if (shortInfo.isValidShortNumber(parsed)) {
  scheme = phonenumber.PhoneNumberFormat.NATIONAL;
} else if (typeof scheme === 'undefined') {
  ...
}
```

In my opinion this is now too many nested ifs, and I would pull it out as a separate function, eg:

```js
function getScheme(given) {
  if (shortInfo.isValidShortNumber(parsed)) {
    return phonenumber.PhoneNumberFormat.NATIONAL;
  }
  if (given) {
    return given;
  }
  if (parsed.getCountryCode() + '' === countryCode) {
    return phonenumber.PhoneNumberFormat.NATIONAL;
  }
  return phonenumber.PhoneNumberFormat.INTERNATIONAL;
}
```

What do you think?

Reply:
Sounds good